src/Adapter/Cart/QueryHandler/GetCartInformationHandler.php

```diff
@@ -238,4 +271,124 @@ private function fetchCartDeliveryOptions(array $deliveryOptionsByAddress, int $
         //make sure array is not associative
         return array_values($deliveryOptions);
     }
+
+    /**
+     * @param array $legacySummary
+     * @param Currency $currency
+     * @param Cart $cart
+     *
+     * @return CartInformation\CartSummary
+     *
+     * @throws LocalizationException
+     */
+    private function extractSummaryFromLegacySummary(array $legacySummary, Currency $currency, Cart $cart): CartSummary
+    {
+        $cartId = (int) $cart->id;
+        $discount = $this->locale->formatPrice($legacySummary['total_discounts_tax_exc'], $currency->iso_code);
+
+        if (0 !== (int) $legacySummary['total_discounts_tax_exc']) {
+            $discount = '-' . $discount;
+        }
+
+        $orderMessage = '';
+        if ($message = Message::getMessageByCartId($cartId)) {
+            $orderMessage = $message['message'];
+        }
+
+        return new CartSummary(
+            $this->locale->formatPrice($legacySummary['total_products'], $currency->iso_code),
+            $discount,
+            $this->locale->formatPrice($legacySummary['total_shipping_tax_exc'], $currency->iso_code),
+            $this->locale->formatPrice($legacySummary['total_tax'], $currency->iso_code),
+            $this->locale->formatPrice($legacySummary['total_price'], $currency->iso_code),
+            $this->locale->formatPrice($legacySummary['total_price_without_tax'], $currency->iso_code),
+            $orderMessage,
+            $this->contextLink->getPageLink(
+                'order',
+                false,
+                (int) $cart->id_lang,
+                http_build_query([
+                    'step' => 3,
+                    'recover_cart' => $cartId,
+                    'token_cart' => md5(_COOKIE_KEY_ . 'recover_cart_' . $cartId),
+                ])
+            )
+        );
+    }
+
+    /**
+     * Provides product customizations data
+     *
+     * @param Cart $cart
+     * @param array $product the product array from legacy summary
+     *
+     * @return CartInformation\Customization|null
+     */
+    private function getProductCustomizedData(Cart $cart, array $product): ?CartInformation\Customization
```

Comment:
```suggestion
    private function getProductCustomizedData(Cart $cart, array $product): ?Customization
```

with the appropriate `use` statemet